doc/conf.py

```diff
@@ -47,12 +47,23 @@
     "add_toctree_functions",
     "sphinx-prompt",
     "sphinxext.opengraph",
+    "matplotlib.sphinxext.plot_directive",
 ]
 
 # this is needed for some reason...
 # see https://github.com/numpy/numpydoc/issues/69
 numpydoc_class_members_toctree = False
 
+# Produce `plot::` directives for examples that contain `import matplotlib` or
```

Comment:
Bumping the matplotlib version for nicer plotting in docstrings does not look like a strong enough reason. A compromise would be to only generate the plots for newer version of matplotlib:

```python
# support for `plot::` directives in sphinx 3.2 require matplotlib 3.1.0
if parse_version(mpl.__version__) > parse_version("3.1.0"):
    extensions.append("matplotlib.sphinxext.plot_directive")
```

This will still generate the plots for the hosted docs.

Reply:
It is fine with me.

This said, I am not sure that I would encourage anymore to not have the latest major revision of a Python package -> pandas 1.x, matplotlib 3.x. But for sure this is not the debate here :)